mis_builder_cash_flow/report/mis_cash_flow.py

```diff
@@ -0,0 +1,132 @@
+from odoo import api, fields, models, tools
+from psycopg2.extensions import AsIs
+
+
+class MisCashFlow(models.Model):
+
+    _name = 'mis.cash_flow'
+    _description = 'MIS Cash Flow'
+    _auto = False
+
+    # line_type = fields.Char()
+    line_type = fields.Selection(
+        [('forecast_line', 'Forecast Line'), ('move_line', 'Journal Item')],
+        index=True,
+        readonly=True,
+    )
+    name = fields.Char(
+        readonly=True,
+    )
+    account_id = fields.Many2one(
+        comodel_name='account.account',
+        string='Account',
+        auto_join=True,
+        index=True,
+        readonly=True,
+    )
+    move_line_id = fields.Many2one(
+        comodel_name='account.move.line',
+        string='Journal Item',
+        auto_join=True,
+        readonly=True,
+    )
+    company_id = fields.Many2one(
+        comodel_name='res.company',
+        string='Company',
+        auto_join=True,
+        readonly=True,
+        index=True,
+    )
+    credit = fields.Float(
+        readonly=True,
+    )
+    debit = fields.Float(
+        readonly=True,
+    )
+    date = fields.Date(
+        readonly=True,
+        index=True,
+    )
+    reconciled = fields.Boolean(
+        readonly=True,
+    )
+    full_reconcile_id = fields.Many2one(
+        'account.full.reconcile',
+        string="Matching Number",
+        readonly=True,
+        index=True,
+    )
+    user_type_id = fields.Many2one(
+        'account.account.type',
+        auto_join=True,
+        readonly=True,
+        index=True,
+    )
+
+    @api.model_cr
+    def init(self):
+        """ TODO find another alternative instead of multiplying
+        aml.id * 10000.
+        We do it that way because if we use "SELECT ROW_NUMBER() OVER()..."
+        the performance was very poor (due to the union of two tables I think)
```

Comment:
I see that they want to avoid having the same ID from the join of the 2 tables, but there's a better option: put here `-aml.id`. You can use negative values for the ID, and this won't conflict for sure, unless you put a third table in the SQL :wink:

Reply:
Great!! I didn't know about the negative id! Thanks! I was becoming crazy looking for a better solution!